newIDE/app/src/MainFrame/index.js

```diff
@@ -3867,17 +3880,47 @@ const MainFrame = (props: Props) => {
           onLaunchPreview={
             hotReloadPreviewButtonProps.launchProjectDataOnlyPreview
           }
-          onClose={options => {
+          onClose={async options => {
+            if (hasUnsavedChanges) {
+              const response = await showConfirmation({
+                title: t`Leave the customization?`,
+                message: t`Do you want to quit the customization? All your changes will be lost.`,
+                confirmButtonLabel: t`Leave`,
+              });
+
+              if (!response) {
+                return;
+              }
+            }
+
             setQuickCustomizationDialogOpenedFromGameId(null);
-            if (options && options.tryAnotherGame) {
-              // Close the project so the user is back at where they can chose a game to customize
-              // which is probably the home page.
-              closeProject();
-              openHomePage();
+            closeProject();
+            openHomePage();
+            if (!hasUnsavedChanges) {
+              navigateToRoute('build');
             }
           }}
           onlineWebExporter={quickPublishOnlineWebExporter}
-          onSaveProject={saveProject}
+          onSaveProject={async () => {
+            // Automatically save project to the cloud if no provider is set.
+            const storageProvider = getStorageProvider();
+            console.log('storageProvider', storageProvider);
+            if (
```

Comment:
should we always save in cloud and not look at the current storage provider?

Reply:
Might be safer to do what you did, but also this introduces some complexity (an additional "if" which is a bit weird for the reader).

I would say:
- Always save in cloud
- Keep the if before so that "if storage provider is not empty and not url storage provider and not cloud, console.error that there is something fishy happening - but still continue"